nomad/structs/structs.go

```diff
@@ -3864,6 +3947,20 @@ type Allocation struct {
 	CreateTime int64
 }
 
+// Index returns the index of the allocation. If the allocation is from a task
+// group with count greater than 1, there will be multiple allocations for it.
+func (a *Allocation) Index() uint {
```

Comment:
We can add it later. Don't want to go through the backwards compatibility exercise in this PR.